remotespark/controllerwidget/abstractmenuwidget.py

```diff
@@ -0,0 +1,40 @@
+from ipywidgets import Box
+
+from remotespark.utils.ipythondisplay import IpythonDisplay
+from remotespark.utils.ipywidgetfactory import IpyWidgetFactory
+
+
+class AbstractMenuWidget(Box):
+    def __init__(self, spark_controller, ipywidget_factory=None, ipython_display=None,
+                 nested_widget_mode=False, testing=False, **kwargs):
+        kwargs['orientation'] = 'vertical'
+
+        if not testing:
+            super(AbstractMenuWidget, self).__init__((), **kwargs)
+
+        self.spark_controller = spark_controller
+
+        if ipywidget_factory is None:
+            ipywidget_factory = IpyWidgetFactory()
+        self.ipywidget_factory = ipywidget_factory
+
+        if ipython_display is None:
+            ipython_display = IpythonDisplay()
+        self.ipython_display = ipython_display
+
+        self.children = []
```

Comment:
This is an attribute that needs to be overridden by a subclass, right?  I'm wondering if you should not set this here so it will be an error if it's accessed and not implemented by the subclass.